### PR TITLE
chore: update RFC process

### PR DESCRIPTION
**Pull request by eladb**
This PR includes changes to our RFC process & template to reflect recent discussions in the CDK team on improvements to the process.

* Updates RFC template:
  * Added a "README" section at the top instead of "Basic Example"
  * Added an "Implementation Section" at the bottom with "Intentionally left blank"

* Refactored lifecycle to account for planning & implementation, so now the process is:

```
proposed 
  => review (was 'pending') 
    => final-comments 
      => approved (was 'ready') 
        => planning (new) 
          => implementing (new) 
            => done (was 'resolved')
```

* Added 'rejected' state
* Simplified README
* Cleaned up PR template
* Update the label names in the renderer

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -169,134 +137,116 @@ Some changes do not require an RFC:
 If you're not sure whether your change requires an RFC, feel free to create an
 issue and ask.
 
-## What the process is
+## RFC Process
 
-In short, to get a major feature added to CDK/JSII, one usually first gets the
-RFC merged into the RFC repo as a markdown file. At that point the RFC is
-'ready' and may be implemented with the goal of eventual inclusion into
+In short, to get a major feature added to CDK/JSII, one usually writes an RFC
```
Comment (njlynch) [inline README.md:142]:
```suggestion
In short, to get a major feature added to CDK/jsii, one usually writes an RFC 
```

There are a few other instances of the upper-case JSII in the rest of the doc, if you're making this change.

File: README.md [hunk 2]
```diff
@@ -169,134 +137,116 @@ Some changes do not require an RFC:
...
-## What the process is
+## RFC Process
 
-In short, to get a major feature added to CDK/JSII, one usually first gets the
-RFC merged into the RFC repo as a markdown file. At that point the RFC is
-'ready' and may be implemented with the goal of eventual inclusion into
+In short, to get a major feature added to CDK/JSII, one usually writes an RFC 
+as a markdown file and gets it approved and merged into the RFC repo. At that point the RFC is
+'approved' and may be implemented with the goal of eventual inclusion into
 CDK/JSII.
```
Comment (njlynch) [inline README.md:145]:
```suggestion
'approved' and may be implemented into CDK/jsii.
```

The "implemented with the goal of eventual inclusion" feels a bit "squishy" to me; if it's implemented and meets the design spec of the RFC, we'll work with the implementor to integrate it. Or is the intent to capture that there may be lag between "contributor implements" and "reviewed, pushed, and live in all published repos"?

File: README.md [hunk 3]
```diff
@@ -169,134 +137,116 @@ Some changes do not require an RFC:
...
   the RFC will receive design feedback from the core team and the larger
   community, and the author should be prepared to make revisions in response.
-  - Update the tracking issue with a link to the RFC PR.
-- Build consensus and integrate feedback. RFCs that have broad support are much
-  more likely to make progress than those that don't receive any comments.
-- Eventually, the team will decide whether the RFC is a candidate for inclusion
+6. Update the tracking issue with a link to the RFC PR.
+7. **Advertise** your RFC amongst stakeholders via social channels (e.g. twitter)
+  and your team. Build consensus and integrate feedback. RFCs that have broad support 
+  are much more likely to make progress than those that don't receive any comments.
```
Comment (njlynch) [inline README.md:163]: \+ Nice; good addition!

File: README.md [hunk 4]
```diff
@@ -169,134 +137,116 @@ Some changes do not require an RFC:
...
+8. Eventually, the team will decide whether the RFC is a candidate for inclusion
   in CDK/JSII.
```
Comment (njlynch) [inline README.md:165]:
```suggestion
  in CDK/jsii.
```

File: README.md [hunk 5]
```diff
@@ -169,134 +137,116 @@ Some changes do not require an RFC:
...
 generally be the ones updating the RFCs state in the tracking issue as it moves
 through the process. They can decide when a final comment period is triggered.
-
-On the other hand, if the submitter is a core team member, they will identify
+>
+> On the other hand, if the submitter is a core team member, they will identify
 another core team member, with consent, as their 'champion'. The champion would
 be the first contact for brainstorming, process and reviews. The core team
 would defer to the champion to do the first few rounds of reviews, after which
 the rest of the team should be engaged.
```
Comment (njlynch) [inline README.md:193]: Does the core team member decide on final comment period in this case, or the champion?

File: README.md [hunk 6]
```diff
@@ -169,134 +137,116 @@ Some changes do not require an RFC:
...
+  are much more likely to make progress than those that don't receive any comments.
+8. Eventually, the team will decide whether the RFC is a candidate for inclusion
   in CDK/JSII.
-- RFCs that are candidates for inclusion in CDK/JSII will enter a "final comment
-  period" lasting 3 calendar days. The beginning of this period will be signaled
+9. RFCs that are candidates for inclusion in CDK/JSII will enter a "**final comment
+  period**" lasting 3 calendar days. The beginning of this period will be signaled
   by a team member adding a comment and label on the RFCs pull request.
-- An RFC can be modified based upon feedback from the team and community.
+10. An RFC can be modified based upon feedback from the team and community.
```
Comment (njlynch) [inline README.md:169]: Is this specifically during the final comment period, or at any time (e.g., after approval)?

File: tools/rfc-render/render-rfc-table.js [hunk 7]
```diff
@@ -3,11 +3,14 @@ const fs = require('fs').promises;
...
+  'status/implementing': { },
+  'status/planning': { },
+  'status/approved': { },
   'status/final-comment-period': { }, 
-  'status/pending': { },
+  'status/review': { },
   'status/proposed': { },
-  'status/resolved': { },
+  'status/done': { },
+  'status/rejected': { },
```
Comment (njlynch) [inline tools/rfc-render/render-rfc-table.js:13]: Nitpick: Does ordering matter here? Logically, I'd either arrange by alpha (order doesn't matter) or chronologically (proposed, review, fcp...).

Comment (eladb) [reply tools/rfc-render/render-rfc-table.js:13]: Yes, ordering matters and intentional, sorted by when it will be delivered (chronological in reverse order). The top items are the ones actually being worked on, then the ones in planning, approved, etc. I put the "done" items in the end because they are less interesting in this list.

Comment (eladb) [reply tools/rfc-render/render-rfc-table.js:13]: I'll add a comment

File: README.md (hunk 5 above)
Comment (eladb) [reply README.md:192]: The champion is the core team members. Let me rephrase.

File: README.md (hunk 6 above)
Comment (eladb) [reply README.md:168]: It can be modified at any time. If it is modified after it was merged, a new PR will be needed (I'll add).

File: README.md [hunk 8]
```diff
@@ -139,57 +139,63 @@ issue and ask.
...
-13. At some point, someone will pick up the RFC for implementation. For major features
-  this usually requires devising a detailed implementation plan. To that end, submit
-  an **additional PR** on the RFC doc that fills in the "Implementation Plan" section.
+    Significant modifications may trigger a new final comment period. An RFC can
+    also be modified after it has been merged and approved, in which case a new
+    PR will be submitted with the modification, like any other code.
+11. An RFC may be **rejected** by the team after public discussion has settled
+    and comments have been made summarizing the rationale for rejection. A
+    member of the team will then close the PR and issue.
+12. An RFC may be accepted at the close of its final comment period. A team
```
Comment (njlynch) [inline README.md:178]:
```suggestion
12. An RFC may be **accepted** at the close of its final comment period. A team
```

Might be worth bolding the "accepted" since you bolded the "rejected".

File: 0000-template.md [hunk 9]
```diff
@@ -5,26 +5,21 @@ rfc pr: (leave this empty)
 related issue: (tracking issue number)
 ---
 
-<!-- replace the blockquoted sections with your content -->
 # Summary
 
 > Brief description of the feature.
 
+# README
```
Comment (nija-at) [inline 0000-template.md:12]: Nice! I like this.

File: 0000-template.md [hunk 10]
```diff
@@ -90,3 +85,12 @@ related issue: (tracking issue number)
 >
 > If you have tried and cannot think of any future possibilities, you may simply
 > state that you cannot think of anything.
+
+# Implementation Plan
```
Comment (nija-at) [inline 0000-template.md:89]: Implementation plans change constantly. I'd rather not put those details in the RFC.

How about we change this title to 'Changes Required' or something along those lines, to summarize what components of the system need to change and how (briefly).

This will keep the RFC nice and clean, focusing only on the design and CX elements of the feature.

File: README.md [hunk 11]
```diff
@@ -160,143 +128,131 @@ first.
...
+11. An RFC may be **rejected** by the team after public discussion has settled
+    and comments have been made summarizing the rationale for rejection. A
+    member of the team will then close the PR and issue.
+12. An RFC may be **accepted** at the close of its final comment period. A team
+    member will merge the RFCs associated pull request, at which point the RFC
+    will become 'approved'.
+13. At some point, someone will pick up the RFC for implementation. For major
+    features this usually requires devising a detailed implementation plan. To
+    that end, submit an **additional PR** on the RFC doc that fills in the
+    "Implementation Plan" section.
```
Comment (nija-at) [inline README.md:184]: Related to my comment above. 

How about create a tracking issue or GH project for every RFC that is accepted which will then hold the implementation plan and its iteration. For starters, it can be brief or high level.

File: 0000-template.md (hunk 9 above)
Comment (eladb) [reply 0000-template.md:12]: :-D

File: 0000-template.md (hunk 10 above)
Comment (eladb) [reply 0000-template.md:89]: One of the goals of this addition is to introduce a formal step in which we review an implementation plan, and I didn't want to introduce yet another document just for that. This is why the template indicates that this should only be filled out when the RFC is ready to be implemented.

I'll revise this slightly to offer users to reference a GitHub Project Board or reference a separate document that includes the implementation plan in order to allow a bit more flexibility.

Would that work?

File: README.md (hunk 11 above)
Comment (eladb) [reply README.md:184]: I'll suggest a project board in the template.

File: 0000-template.md (hunk 10 above)
Comment (eladb) [reply 0000-template.md:89]: Slightly reworded. I still think there is value in adding a link to the plan once planning is performed.

File: README.md (hunk 11 above)
Comment (eladb) [reply README.md:184]: Added

File: README.md [hunk 12]
```diff
@@ -75,20 +77,13 @@ existing proposals.
...
-This document is a lot of information about process thats meant to help guide.
-It is not a set of rules that need to be strictly applied. It is designed to
-help contributors (and thats you!) become more involved with the tools that they
-rely on. All efforts to contribute are encouraged and appreciated.
-
 ## What is an RFC?
 
 An RFC is a document that proposes and details a change or addition to the CDK,
-JSII, and other related tooling. It also is a process for reviewing and
+jsii, and other related tooling. It also is a process for reviewing and
```
Comment (MrArnoldPalmer) [inline README.md:83]: The only nitpick I have is all of our other acronyms we capitalize, ie "CDK", "AWS" etc so for consistency I'd argue it should be "JSII" though obviously not something important.

File: 0000-template.md [hunk 13]
```diff
@@ -5,26 +5,23 @@ rfc pr: (leave this empty)
...
 # Summary
 
 > Brief description of the feature.
 
+# README
+
+> Work backwards from the README: pretend that this feature is already
+> implemented and write the README section. It will help you think about your
+> proposal from a user experience point of view. If a README format does not
+> make sense for your proposal, write a "Press Release" which announces your new
```
Comment (MrArnoldPalmer) [inline 0000-template.md:17]: I worry about people reading this and not quite understanding what we are looking for. Can we link an example specifically for the "press release" format?

I also don't love the idea of their being 2 potential formats, README or PR depending on the feature. I'm kinda in favor of just picking "press release" with some predetermined sections that we define to help guide RFC authors.

**Comment by MrArnoldPalmer** [comment]
And I missed it. RE: "press release vs readme" I think we can collect some feedback here going forward.

File: 0000-template.md (hunk 13 above)
Comment (eladb) [reply 0000-template.md:17]: Let's give it a bit of time if that's okay with you and see if we need to make this a bit more concrete. We will gather some examples so we can reference.

File: README.md (hunk 12 above)
Comment (eladb) [reply README.md:83]: We've been using `jsii` lowercase across the board. If @RomainMuller and you prefer that we change that I think we should probably change it everywhere and not just here.

File: 0000-template.md [hunk 14]
```diff
@@ -90,3 +87,16 @@ related issue: (tracking issue number)
 >
 > If you have tried and cannot think of any future possibilities, you may simply
 > state that you cannot think of anything.
+
+# Implementation Plan
+
+INTENTIONALLY LEFT BLANK: an implementation plan will be added when the RFC is
+scheduled for implementation.
+
```
Comment (NetaNir) [inline 0000-template.md:95]: maybe "should be added after the RFC is approved"?
